crates/lsp/src/lsp.rs

```diff
@@ -846,6 +846,21 @@ impl LanguageServer {
                     color_provider: Some(DocumentColorClientCapabilities {
                         dynamic_registration: Some(true),
                     }),
+                    folding_range: Some(FoldingRangeClientCapabilities {
+                        dynamic_registration: Some(false),
+                        range_limit: None,
+                        line_folding_only: Some(false),
+                        folding_range_kind: Some(FoldingRangeKindCapability {
+                            value_set: Some(vec![
+                                FoldingRangeKind::Comment,
+                                FoldingRangeKind::Imports,
+                                FoldingRangeKind::Region,
+                            ]),
+                        }),
+                        folding_range: Some(FoldingRangeCapability {
+                            collapsed_text: Some(true),
```

Comment:
This is tricky: now, by default, Zed displays an ellipsis symbol (`…`) on the fold place, and lays out the string the same way.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_foldingRange says that
```
		* If set, the client signals that it supports setting collapsedText on
		* folding ranges to display custom labels instead of the default text.
```

meaning that it may cause quite a change to the line layouting.

I suspect there might be issues, so if we follow this way and keep that enabled, we'd need to account for that in the corresponding block rendering: https://github.com/zed-industries/zed/blob/cb0e85cf993a0fd94a4e44956a11f27ccf11ff7a/crates/editor/src/editor.rs#L1796-L1827